src/python/pants/backend/jvm/tasks/ivy_task_mixin.py

```diff
@@ -101,15 +101,32 @@ def implementation_version(cls):
     return super(IvyTaskMixin, cls).implementation_version() + [('IvyTaskMixin', 4)]
 
   @memoized_property
-  def ivy_cache_dir(self):
-    """The path of the ivy cache dir used for resolves.
+  def ivy_repository_cache_dir(self):
+    """The path of the ivy cache dir used for repository cache data.
 
     :API: public
```

Comment:
Since this is a public method (... I have no idea why, sorry!) ... you'll need to preserve an `@deprecated` copy with the old name.

You might consider not making the new methods `:API: public`, to avoid this issue in the future.